File: ballast-core/dotnet/src/dependency-injection/Options.cs
using System.Collections.Generic;
using Ballast.Core.Domain.Models;
using Ballast.Core.Application.Services.Impl;

namespace Ballast.Core.DependencyInjection 
{
    public interface IBallastCoreOptions : IDomainGameServiceOptions
    { 
        bool UseDomain { get; set; }
        bool UseLocalEventBus { get; set; }
    }

    public class BallastCoreOptions : IBallastCoreOptions
    {

        public bool UseDomain { get; set; }
        public bool UseLocalEventBus { get; set; }
        public string DefaultBoardType { get; set; }
        public string DefaultTileShape { get; set; }
        public int? DefaultBoardSize { get; set; }
        public double? DefaultLandToWaterRatio { get; set; }
        public IEnumerable<string> DefaultVessels { get; set; }

        public BallastCoreOptions() 
        { 
            UseDomain = true;
            UseLocalEventBus = true;
        }

    }
}

File: ballast-core/npm/src/dependency-injection/options.ts
import { IDomainGameServiceOptions } from "../application/services/impl/domain-game-service"

export interface IBallastCoreOptions extends IDomainGameServiceOptions {
    useDomain: boolean;
    useLocalEventBus: boolean;
}

export class BallastCoreOptions implements IBallastCoreOptions { 

    public useDomain: boolean;
    public useLocalEventBus: boolean;
    public defaultBoardType: string | null;
    public defaultTileShape: string | null;
    public defaultBoardSize: number | null;
    public defaultLandToWaterRatio: number | null;
    public defaultVessels: string[] | null;

    public constructor() {
        this.useDomain = true;
        this.useLocalEventBus = true;
        this.defaultBoardType = null;
        this.defaultTileShape = null;
        this.defaultBoardSize = null;
        this.defaultLandToWaterRatio = null;
        this.defaultVessels = [];
    }
    
}